src/compile/data/assemble.ts

```diff
@@ -156,6 +156,19 @@ function makeWalkTree(data: VgData[]) {
    * Recursively walk down the tree.
    */
   function walkTree(node: DataFlowNode, dataSource: VgData) {
+    if (node instanceof ParseNode) {
+      if (node.parent instanceof SourceNode && !node.parent.parent)  {
+        // If its parent is a root source node, use normal format parse
+        dataSource.format = {
```

Comment:
I rebased #2168 to #2174 and found that even though a parse node's parent is a source, it does not mean that the `dataSource` we are assembling here is the sourceNode's data source.  In #2168, for box_plot example and several others, `dataSource` here are new data sources that depends on the sourceNode's data source.